MapboxNavigation/InstructionPresenter.swift

```diff
@@ -63,105 +63,100 @@ class InstructionPresenter {
+            case let .exitCode(representation) where 0...1 ~= index:
+                guard let exitString = self.attributedString(forExitRepresentation: representation, maneuverDirection: instruction.maneuverDirection!, dataSource: dataSource, cacheKey: component.cacheKey!) else { fallthrough }
                 build(component, [exitString])
                 
             //if it's a delimiter, skip it if it's between two shields.
             case .delimiter where isShield(componentBefore) && isShield(componentAfter):
                 continue
                 
             //If we have an icon component, lets turn it into a shield.
-            case .image:
-                if let shieldString = attributedString(forShieldComponent: component, repository: imageRepository, dataSource: dataSource, onImageDownload: onImageDownload) {
+            case let .image(imageRepresentation, textRepresentation):
+                if let shieldString = attributedString(forShieldComponent: imageRepresentation, repository: imageRepository, dataSource: dataSource, cacheKey: component.cacheKey!, onImageDownload: onImageDownload) {
                     build(component, [joinString, shieldString])
-                } else if let genericShieldString = attributedString(forGenericShield: component, dataSource: dataSource) {
+                } else if let genericShieldString = attributedString(forGenericShield: textRepresentation, dataSource: dataSource, cacheKey: component.cacheKey!) {
                     build(component, [joinString, genericShieldString])
                 } else {
                     fallthrough
                 }
                 
-            //Otherwise, process as text component.
-            default:
-                guard let componentString = attributedString(forTextComponent: component, dataSource: dataSource) else { continue }
+            case let .text(textRepresentation), let .delimiter(textRepresentation):
+                let componentString = NSAttributedString(string: textRepresentation.text, attributes: attributes(for: dataSource))
                 build(component, [joinString, componentString])
+            
             }
         }
         
         assert(processedComponents.count == strings.count, "The number of processed components must match the number of attributed strings")
         return (components: processedComponents, attributedStrings: strings)
     }
 
-    func attributedString(forExitComponent component: VisualInstructionComponent, maneuverDirection: ManeuverDirection, dataSource: DataSource) -> NSAttributedString? {
-        guard component.type == .exitCode, let exitCode = component.text else { return nil }
+    func attributedString(forExitRepresentation representation: VisualInstruction.Component.TextRepresentation, maneuverDirection: ManeuverDirection, dataSource: DataSource, cacheKey: String) -> NSAttributedString? {
```

Comment:
The new VisualInstruction.Component enumeration makes it tempting to consolidate these “attributed string for” methods into a single method with a switch statement.